### ISSUE TITLE
Mutators for List<T>

### ISSUE THREAD

**Issue by yakivyusin** [stryker-mutator/stryker-net#1331]
**Is your feature request related to a problem? Please describe.**
`List<T>` is the one from the most using structures, would be good to have mutators for its methods.

**Describe the solution you'd like**
| Original                      | Mutated                       |
|-------------------------------|-------------------------------|
| `Add(T)`                      | `Remove(T)`                   |
| `Remove(T)`                   | `Add(T)`                      |
| `IndexOf(T)`                  | `LastIndexOf(T)`              |
| `LastIndexOf(T)`              | `IndexOf(T)`                  |
| `Find(Predicate<T>)`          | `FindLast(Predicate<T>)`      |
| `FindLast(Predicate<T>)`      | `Find(Predicate<T>)`          |
| `FindIndex(Predicate<T>)`     | `FindLastIndex(Predicate<T>)` |
| `FindLastIndex(Predicate<T>)` | `FindIndex(Predicate<T>)`     |
| `Sort()`                      | nothing                       |
| `TrueForAll(Predicate<T>)`    | `Exists(Predicate<T>)`        |
| `Exists(Predicate<T>)`        | `TrueForAll(Predicate<T>)`    |

If it will be approved, I may start working on the implementation.

**Comment by richardwerkman** [comment]
Good idea! We would have to solve this just like we did with linq methods. We can only look at the method names and not to the type of the variable. So we can't be sure the type will be List. Luckily our rollback mechanism will just remove any mutations that are invalid.

We could add these mutations to a higher mutation level as they give extra insight but don't seems required to me. I suggest giving the mutation the level of `Advanced`. @Mobrockers agreed?

**Comment by Mobrockers** [comment]
I think we'll have to see how much compilation time this adds to a real life project to decide if it should become Advanced or Complete.
